src/sdk/data-api.ts

```diff
@@ -217,4 +256,22 @@ export class DataApiImpl implements DataApi {
       return subrequestResults.then((keyValues) => new Map(keyValues));
     });
   }
+
+  private validate_response(response, validator) {
+    if (typeof response !== "object" || !validator(response)) {
+      throw new Error("Could not parse API response as JSON!");
+    }
+  }
+
+  private handle_bad_response(error) {
```

Comment:
can you add a comment here like

```
// jsforce sets response body into `message` instead of `content`, so the output would not be helpful
```

we should also file a bug report (or, better yet, make PR) on the jsforce repo